src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/killaura/features/KillAuraVelocityHit.kt

```diff
@@ -0,0 +1,111 @@
+package net.ccbluex.liquidbounce.features.module.modules.combat.killaura.features
+
+import com.google.common.collect.Lists
+import net.ccbluex.liquidbounce.config.types.ToggleableConfigurable
+import net.ccbluex.liquidbounce.event.events.PacketEvent
+import net.ccbluex.liquidbounce.event.events.TransferOrigin
+import net.ccbluex.liquidbounce.event.sequenceHandler
+import net.ccbluex.liquidbounce.event.tickHandler
+import net.ccbluex.liquidbounce.features.module.modules.combat.killaura.ModuleKillAura
+import net.ccbluex.liquidbounce.utils.client.Chronometer
+import net.ccbluex.liquidbounce.utils.client.PacketQueueManager
+import net.minecraft.network.packet.s2c.play.*
+
+object KillAuraVelocityHit : ToggleableConfigurable(ModuleKillAura, "VelocityHit", false) {
+
+    val extendRange by float("ExtendRange", 1.0f, 0.1f..2.0f)
+    private val whenLag by boolean("OnlyWhileLagging", false)
+
+    private var considerVelocityHit = false
+    private var damageReceived = false
+    private var onGroundTicks = 0
+    private var isPossible = false
+
+    private var timer = Chronometer()
+    private var lastPacketTime = Lists.newLinkedList<Long>()
+
+    private const val SAMPLE_SIZE = 10
+
+    val isVelocityHitPossible
+        get() = super.running && isPossible
+
+    @Suppress("unused")
+    private val packetHandler = sequenceHandler<PacketEvent>(priority = 1) { event ->
+        val packet = event.packet
+
+        if (event.origin == TransferOrigin.RECEIVE) {
+            addRecentPacketTime()
+        }
+
+        when {
+            packet is EntityDamageS2CPacket && packet.entityId == player.id -> {
+                damageReceived = true
+            }
+            packet is EntityVelocityUpdateS2CPacket && packet.entityId == player.id && damageReceived -> {
+                considerVelocityHit = true
+            }
```

Comment:
you have implemented a state machine, but you use two separate flags to keep track of what state the state machine is currently in.

I'd recommend using an enum for the states, for example:
```kotlin
enum PacketListenerState {
    IDLE,
    /**
     * The player received damage, but no velocity packet was received (yet)
     */
    DAMAGE_RECEIVED,
    /**
     * A hit with velocity was received.
     */
    VELOCITY_DAMAGE_RECEIVED
}